### DISCUSSION TITLE
GitOps with Hetzner-k3s? #Terraform, #Crossplane

### DISCUSSION THREAD

**Discussion by michaelfresco** [vitobotta/hetzner-k3s#243] [General]
I looked at Crossplane, and this tool seems really cool. Crossplane makes sure
that your Kubernetes cluster looks exactly like you specified it in some Git
repo. So you just put your `cluster.yaml` under version control, git commit
it, push it to some repo, and Crossplane will "talk" to your cloud provider's API,
and create a cluster. It functions in a way that is very similar to Terraform.

If we want to do something similar with Hetzner k3s, how would you approach this?
Would you manually create a Docker container, which has Hetzner-k3s in it, and
add some sort of loop?

**Comment by michaelfresco** [comment]
More a Crossplane thing, I'll continue searching for solutions.

**Comment by vitobotta** [comment]
Hi, I have never tried Crossplane nor looked much into it yet, so I would have no idea :)
